packages/adapter-next/src/hooks/project-init.ts

```diff
@@ -40,31 +42,141 @@ const createPrismicIOFile = async ({
 		helpers,
 		options,
 	});
+	const hasSrcDirectory = await checkHasSrcDirectory({ helpers });
+	const hasAppRouter = await checkHasAppRouter({ helpers });
 
-	const filePath = helpers.joinPathFromRoot(
-		isTypeScriptProject ? "prismicio.ts" : "prismicio.js",
-	);
+	const filename = isTypeScriptProject ? "prismicio.ts" : "prismicio.js";
+	const filePath = hasSrcDirectory
+		? helpers.joinPathFromRoot("src", filename)
+		: helpers.joinPathFromRoot(filename);
 
 	if (await checkPathExists(filePath)) {
 		return;
 	}
 
+	let createClientContents: string;
+
+	if (hasAppRouter) {
+		if (isTypeScriptProject) {
+			createClientContents = source`
+				/**
+				 * Creates a Prismic client for the project's repository. The client is used to
+				 * query content from the Prismic API.
+				 *
+				 * @param config - Configuration for the Prismic client.
+				 */
+				export const createClient = (config: prismicNext.CreateClientConfig = {}) => {
+					const client = prismic.createClient(repositoryName, {
+						routes,
+						fetchOptions:
+							process.env.NODE_ENV === 'production'
+								? { next: { tags: ['prismic'] }, cache: 'force-cache' }
+								: { next: { revalidate: 5 } },
+						...config,
+					});
+
+					prismicNext.enableAutoPreviews({
+						client,
+						previewData: config.previewData,
+						req: config.req,
+					});
+
+					return client;
+				};
+			`;
+		} else {
+			createClientContents = source`
+				/**
+				 * Creates a Prismic client for the project's repository. The client is used to
+				 * query content from the Prismic API.
+				 *
+				 * @param {prismicNext.CreateClientConfig} config - Configuration for the Prismic client.
+				 */
+				export const createClient = (config = {}) => {
+					const client = prismic.createClient(repositoryName, {
```

Comment:
There are two pieces that change, which is why we see four separate code blocks:

1. If the project uses TypeScript, we use TSDoc and inline the parameter's type. Otherwise, we use JSDoc for both.
2. If the project uses the App Router, we add an App Router-specific `fetchOptions` parameter to `@prismicio/client`.

We could potentially use multiple variables to construct the resulting string, but I felt it would be harder to follow. The current method is certainly longer and the differences may not be obvious between snippets, but I think it ultimately is more maintainable.

Happy to reconsider if you have a strong opinion!